backend/sirius-web-emf/src/test/java/org/eclipse/sirius/web/emf/compatibility/operations/CreateViewOperationHandlerTests.java

```diff
@@ -0,0 +1,286 @@
+import org.eclipse.sirius.web.emf.compatibility.modeloperations.ChildModelOperationHandler;
+import org.eclipse.sirius.web.emf.compatibility.modeloperations.CreateViewOperationHandler;
+import org.eclipse.sirius.web.emf.services.EditingContext;
+import org.eclipse.sirius.web.representations.IStatus;
+import org.eclipse.sirius.web.representations.Success;
+import org.eclipse.sirius.web.representations.VariableManager;
+import org.eclipse.sirius.web.spring.collaborative.diagrams.api.IDiagramContext;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests of the CreateView operation handler.
+ *
+ * @author arichard
+ */
+public class CreateViewOperationHandlerTests {
+    /**
+     *
+     */
+    private static final String AQL = "aql:"; //$NON-NLS-1$
+
+    private static final String VARIABLE_NAME = "myVariableName"; //$NON-NLS-1$
+
+    private static final String CONTAINER_VIEW = "containerView"; //$NON-NLS-1$
+
+    private CreateViewOperationHandler createViewOperationHandler;
+
+    private CreateView createViewOperation;
+
+    private OperationTestContext operationTestContext;
+
+    @BeforeEach
+    public void initialize() {
+        this.operationTestContext = new OperationTestContext();
+
+        ComposedAdapterFactory composedAdapterFactory = new ComposedAdapterFactory();
+        composedAdapterFactory.addAdapterFactory(new EcoreAdapterFactory());
+        composedAdapterFactory.addAdapterFactory(new ReflectiveItemProviderAdapterFactory());
+
+        EPackage.Registry ePackageRegistry = new EPackageRegistryImpl();
+        ePackageRegistry.put(EcorePackage.eINSTANCE.getNsURI(), EcorePackage.eINSTANCE);
+
+        ResourceSet resourceSet = new ResourceSetImpl();
+        resourceSet.setPackageRegistry(ePackageRegistry);
+
+        // @formatter:off
+        DiagramDescription diagramDescription = DiagramDescription.newDiagramDescription(UUID.randomUUID())
+                .label("DiagramDescriptionTest") //$NON-NLS-1$
+                .targetObjectIdProvider(variableManager -> "diagramTargetObjectId") //$NON-NLS-1$
+                .canCreatePredicate(variableManager -> true)
+                .labelProvider(variableManager -> "Diagram") //$NON-NLS-1$
+                .toolSections(List.of())
+                .nodeDescriptions(List.of(this.getNodeDescription(UUID.randomUUID())))
+                .edgeDescriptions(List.of())
+                .build();
+        Diagram diagram = Diagram.newDiagram(UUID.randomUUID())
+                .descriptionId(diagramDescription.getId())
+                .targetObjectId(UUID.randomUUID().toString())
+                .label("DiagramTest") //$NON-NLS-1$
+                .position(Position.at(0, 0))
+                .size(Size.of(100, 100))
+                .nodes(new ArrayList<>())
+                .edges(List.of())
+                .build();
+        IDiagramContext diagramContext = new IDiagramContext() {
```

Comment:
new IDiagramContext.NoOp() (to be created if it does not exist)

Reply:
Done